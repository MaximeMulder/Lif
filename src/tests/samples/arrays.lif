print(Array[Any](0, true, "String"));

let a = Array[Integer](0);
a(0);

print(Array[Integer](2, 3, 4)(0));
print(Array[Integer](2, 3, 4)(2));

let array = Array[Integer](0, 1, 2);
array.append(3);
print(array);

let array = Array[Integer](0, 1, 2);
array.prepend(-1);
print(array);

let array = Array[Integer](0, 1, 2);
array.insert(1, 10);
print(array);

let array = Array[Integer](0, 1, 2);
array.remove(1);
print(array);

let array = Array[Integer](2, 4, 8);
array.append(16, 32, 64);
array.prepend(0, 1);
print(array);
