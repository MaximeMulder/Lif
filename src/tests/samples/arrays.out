[0, true, String]
2
4
[0, 1, 2, 3]
[-1, 0, 1, 2]
[0, 10, 1, 2]
[0, 2]
[0, 1, 2, 4, 8, 16, 32, 64]
